core/modules/layout/includes/block.text.inc

```diff
@@ -91,6 +91,18 @@ class BlockText extends Block {
           ),
         ),
       );
+      $form['convert']['description'] = array(
+       '#type' => 'textfield',
+       '#title' => t('Block description'),
+       '#maxlength' => 128,
+       '#description' => t('Used in administrative interfaces and not shown to end-users.'),
+       '#weight' => 3,
```

Comment:
@Graham-72 no javascript wizardry going on 😄 ..well, it is JS, but not magic. All you need to do is click on the "edit" link next to the auto-generated machine name:

![image](https://user-images.githubusercontent.com/2423362/27031838-44047dd8-4fb5-11e7-98b5-edbd14df5f6b.png)